proposed/2022/globalpackagereference.md

```diff
@@ -0,0 +1,125 @@
+default metadata to ensure that only package assets related to build purposes are used.
+
+**NOTE:** This functionality will only be enabled if a user has opt-ed into [Central Package Management (CPM)](https://docs.microsoft.com/nuget/consume-packages/central-package-management).
+
+```xml
+<Project>
+  <ItemGroup>
+    <GlobalPackageReference Include="Nerdbank.GitVersioning" Version="1.0.0" />
+  </ItemGroup>
+</Project>
+```
+
+### Technical explanation
+<!-- Explain the proposal in sufficient detail with implementation details, interaction models, and clarification of corner cases. -->
+`<GlobalPackageReference />` will just be syntactical sugar for a `<PackageReference />` with the following metadata:
+
+* IncludeAssets = `runtime; build; native; contentfiles; analyzers`
+* PrivateAssets = `All`
+
+This ensures that when these repository-wide packages are consumed, they don't flow to downstream dependencies and only include assets related to
+build.
+
+This will be achieved by copying the `GlobalPackageReference` items to `PackageReference` and `PackageVersion` items:
+```xml
+<Target Name="CollectCentralPackageVersions" Returns="@(PackageVersion)">
+  <ItemGroup Condition="'$(ManagePackageVersionsCentrally)' == 'true' And '$(RestoreEnableGlobalPackageReference)' != 'false'">
+    <!--
+      Add GlobalPackageReference items to the PackageVersion item group with the version.  The PackageReference items are added
+      in the CollectPackageReferences target.
+    -->
+    <PackageVersion Include="@(GlobalPackageReference)" Version="%(Version)" />
+  </ItemGroup>
+</Target>
+
+<Target Name="CollectPackageReferences" Returns="@(PackageReference)" >
+  <ItemGroup Condition="'$(ManagePackageVersionsCentrally)' == 'true' And '$(RestoreEnableGlobalPackageReference)' != 'false'">
+    <!--
+      Add GlobalPackageReference items to the PackageReference item group with no version.  The PackageVersion items are added
+      in the CollectCentralPackageVersions target.
+
+      Global package references only include the same assets as a development dependency (runtime; build; native; contentfiles; analyzers)
+      because those kind of packages are the best candidate for a global package reference.  They are generally packages that
+      extend the build.
+
+      Global package references have all assets private because central package references are generally packages that provide
+      versioning, signing, etc and should not flow to downstream dependencies.  Also, central package references are already
+      referenced by every project in the tree so they don't need to be transitive.
+    -->
+    <PackageReference Include="@(GlobalPackageReference)"
+                      Version=""
+                      IncludeAssets="runtime; build; native; contentfiles; analyzers"
+                      PrivateAssets="All" />
```

Comment:
With the item definition, you can override the GPC's metadata but this removes the ability to that. If we specify GPC with a custom metadata, those will get overriden here.